src/libgeds/TcpTransport.cpp

```diff
@@ -118,6 +118,7 @@ void TcpPeer::cleanup() {
     LOG_DEBUG("Endpoint shutdown: socket: ", tep->sock, " sent: ", tep->tx_bytes,
               " received: ", tep->rx_bytes);
   }
+  endpoints.clear();
```

Comment:
I think I had this delayed until TcpTransport::stop to handle in-flight operations on the shutdown sockets. The shutdown will wakeup the associated epoll waiting thread, which in turn shall complete in error any in-flight rpc.

Reply:
`endpoints` is never cleared.

Reply:
absolutely! you are right. And we are in overall tcp transport shutdown. So let's clear that map
 here!